### PR TITLE
AAC4 Bypasses and more

### PR DESCRIPTION
**Pull request by bestnub**
RotationStrafe-Calculations are not very elegant

### CODE REVIEW

**Comment by kawaiinekololis** [comment]
Why you closed your pull request? Is there any specific reason? 
In case you want to make new pull requests in feature, could you make one pull request for every feature?

**Comment by kawaiinekololis** [comment]
**Approved**
- Added RotationModes … f620cf7
- Added Usefull setting a5a894b
- Improved RotationStrafe f90850e

**Not approved**
- Added Sliders for BlockStacks, EnchantedBooks and buckets … 89b770f
_We don't want to add more item specific options in InventoryCleaner_

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/KillAura.kt [hunk 1]
```diff
@@ -234,12 +234,90 @@ class KillAura : Module() {
...
+        if (calcStrafe > 1f) {
+            calcStrafe *= 0.5f
+        } else if (calcStrafe < 0.9f && calcStrafe > 0.3f) {
+            calcStrafe *= 0.5f
+        }
+        if (calcStrafe < -1f) {
+            calcStrafe *= 0.5f
+        } else if (calcStrafe > -0.9f && calcStrafe < -0.3f) {
+            calcStrafe *= 0.5f
+        }
```
Comment (kawaiinekololis) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/KillAura.kt:318]: The code should be optimized.

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Scaffold.java [hunk 2]
```diff
@@ -241,6 +269,125 @@ public void onPacket(final PacketEvent event) {
...
+            if (calcStrafe > 1F) {
+                calcStrafe *= 0.5F;
+            } else if (calcStrafe < 0.9F && calcStrafe > 0.3F) {
+                calcStrafe *= 0.5F;
+            }
+            if (calcStrafe < -1F) {
+                calcStrafe *= 0.5F;
+            } else if (calcStrafe > -0.9F && calcStrafe < -0.3F) {
+                calcStrafe *= 0.5F;
+            }
```
Comment (kawaiinekololis) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/world/Scaffold.java:367]: The code should be optimized.

**Comment by bestnub** [comment]
I closed it because RotationStrafe Code ist not optimzed, StaticPitch mode in Stafford is not good and the Siders in InventoryCleaner didn't work as expected

**Comment by yorik100** [comment]
> 
> 
> **Approved**
> 
>     * Added RotationModes … [f620cf7](https://github.com/CCBlueX/LiquidBounce/commit/f620cf7c1091cf48db67d57ce5729111018bafa5)
> 
>     * Added Usefull setting [a5a894b](https://github.com/CCBlueX/LiquidBounce/commit/a5a894b7daec8d1b2543f6f123edcde1fd922f5f)
> 
>     * Improved RotationStrafe [f90850e](https://github.com/CCBlueX/LiquidBounce/commit/f90850e162ad828d581421c84734e54529a2e4e7)
> 
> 
> **Not approved**
> 
>     * Added Sliders for BlockStacks, EnchantedBooks and buckets … [89b770f](https://github.com/CCBlueX/LiquidBounce/commit/89b770f5f08c0752e14e38639ebc65cb4678d84e)
>       _We don't want to add more item specific options in InventoryCleaner_

ffs my inventory keeps getting flooded with useless 9827836782468 lava/water buckets and enchant books, it always end up like that and I don't call this clean https://cdn.discordapp.com/attachments/463422722416836628/699649525991735428/unknown.png

**Comment by Turtl-e** [comment]
this code is shit ngl

**Comment by Turtl-e** [comment]
all ur "improved" rotationstrafe code can be done in less than 6 lines of code when u have 72

**Comment by SenkJu** [comment]
If you know so much about good code, why don't you go ahead and re-work this pull request for him?

**Comment by Turtl-e** [comment]
> If you know so much about good code, why don't you go ahead and re-work this pull request for him?

i wanted to make op rotationstrafe but you literally stopped releasing dev builds after implementing onStrafe event into the code and im too lazy to actually build it :v

**Comment by Turtl-e** [comment]
and im not gonna do work for someone, learn how to code, noone will do everything for u (for free ofc)

**Comment by SenkJu** [comment]
What? You can download the latest nightly release from our website right now? Your constant complaining about other's code without ever contributing anything yourself is starting to get on my nerves. Do you ever have anything productive to say?

**Comment by Turtl-e** [comment]
you dont need to bake bread to say that it taste good or shit lol

**Comment by Turtl-e** [comment]
also you had that bot that was sending and i had break for 2/3 months and didnt knew you started to post it on website :v

**Comment by SenkJu** [comment]
> you dont need to bake bread to say that it taste good or shit lol

That's true but what you are doing is not doing anything good for anyone. You are constantly whining about bad code here and bad code there but not providing any solutions. SilentRotationStrafe is actually a very useful feature. The just has to be re-worked by someone with more experience in writing elegant code. 

There is a difference between constructive criticism and just calling something terrible.

**Comment by Turtl-e** [comment]
give me some time, im gonna make that rotationstrafe and share it

**Comment by bestnub** [comment]
Pls review downscaffold

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/KillAura.kt (hunk 1 above)
Comment (superblaubeere27) [reply 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/combat/KillAura.kt:318]: I think this can be done when it is merged

**Comment by kawaiinekololis** [comment]
Not really. The pull request should be separated into smaller pull requests by features, so we can merge them without merging everything.

**Comment by bestnub** [comment]
Is there a good way to do that?

**Comment by Turtl-e** [comment]
smh onStrafe doesn't work correctly in scriptapi and i'm too lazy to download and build lb tbh. Even the simple onStrafe that is already in lb flags. I'm not gonna help u then :v

**Comment by SenkJu** [comment]
> smh onStrafe doesn't work correctly in scriptapi and i'm too lazy to download and build lb tbh. Even the simple onStrafe that is already in lb flags. I'm not gonna help u then :v

It does work exactly the same way as it does in LiquidBounce.
https://github.com/CCBlueX/LiquidBounce/blob/master/1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/script/api/module/ScriptModule.kt#L173

**Comment by superblaubeere27** [comment]
@bestnub I messed up when I tried to fix your PR. Git is a bit weird. We could just merge in the PR's Diff (not the PR's patch as GitHub did), but you would lose your credit which wouldn't be fair IMO. Can you please apply this patch to the master branch? If you want to do this, you have to execute following commands: ```
git checkout -b aacv4_rotation_strafe
git remote add CCBlueX https://github.com/CCBlueX/LiquidBounce.git
git fetch CCBlueX
git reset --hard CCBlueX/master
git apply 81.patch
git add 1.8.9-Forge/src
git commit -m "Added RotationStrafe"
git push <your-remote>```
Then you can create a new PR from that branch. Sorry for the inconvenience

**Comment by bestnub** [comment]
> @bestnub I messed up when I tried to fix your PR. Git is a bit weird. We could just merge in the PR's Diff (not the PR's patch as GitHub did), but you would lose your credit which wouldn't be fair IMO. Can you please apply this patch to the master branch? If you want to do this, you have to execute following commands:
> 
> ```shell
> git checkout -b aacv4_rotation_strafe
> git remote add CCBlueX https://github.com/CCBlueX/LiquidBounce.git
> git fetch CCBlueX
> git reset --hard CCBlueX/master
> git apply 81.patch
> git add 1.8.9-Forge/src
> git commit -m "Added RotationStrafe"
> git push [your-remote]
> ```
> 
> Then you can create a new PR from that branch. Sorry for the inconvenience
Problem:
error: can't open patch '81.patch': No such file or directory

Just merge it as you want. I don't care about credit.

**Comment by superblaubeere27** [comment]
@bestnub I noticed that you went on the contributor list because we merging this PR. 
grafik
Thank you for your contribution ❤️

**Comment by bestnub** [comment]
Cool

**Pull request merged**